doc/content/components/cli.md

```diff
@@ -4,6 +4,86 @@ description: ""
 weight: 7
 ---
 
-The Command-line Interface provides a cross-platform interface for managing components through the command-line.
+The Command-line Interface (CLI) provides a cross-platform interface for managing components through the command-line.
 
 <!--more-->
+
+## Configuration
+
+See the [Configuration Reference]({{< ref "/reference/configuration" >}}) for details on how the CLI can be configured. The different options are detailed in the [CLI Configuration Reference]({{< ref "/reference/configuration/cli.md" >}}).
+
+## Help for Commands
+
+The CLI comes with builtin documentation. This documentation can be printed by adding the `-h` or `--help` flag to any command.
+
+## Command Aliases
+
+Many CLI commands names have aliases, which are different spellings or shorter versions of the same command. For instance, `applications` commands can also be written as `application`, `apps`, `app` or even `a`. This could save you some typing if you frequently use the CLI.
+
+> Tip: For even less typing you could also register an alias for `ttn-lw-cli` itself (`alias tt=ttn-lw-cli`).
+
+## Input and Output Formats
+
+By default the CLI outputs results as one or more JSON objects. These JSON objects are equivalent to the JSON objects that would be returned when using the HTTP API of The Things Stack. Some commands accept one or more JSON objects as input. These JSON objects are equivalent to the JSON objects that can be sent to the HTTP API. This way of handling input and output is used by the [End Device Templates feature]({{< ref "/reference/end-device-templates" >}}).
+
+The `--output-format` flag allows you to specify a [Go template](https://golang.org/pkg/text/template/) that is executed for the result(s) of a command. The example below executes lists applications and outputs their IDs and names:
+
+```bash
+$ ttn-lw-cli applications list --name --output-format "{{ .ApplicationID }}: {{ .Name }}"
+```
+
+## Login and Logout
+
+The `login` command starts the OAuth authorization flow with the configured OAuth server. By default this makes use of an endpoint on `localhost` for the OAuth callback. If you are not running the CLI on the machine that is `localhost`, you can add the flag `--callback=false` to the `login` command. This will disable the callback endpoint, and instead ask you to copy/paste the authorization code that you will receive from the OAuth server.
+
+It is also possible to use an API key instead of OAuth. To do this, add the flag `--api-key=NNSXS.AAAA.BBBB` to the `login` command (replacing `NNSXS.AAAA.BBBB` with your API key).
+
+Logging out of the CLI is done with the `logout` command, which will automatically invalidate the OAuth access token that was in use.
+
+## Entity Management
+
+In most cases the CLI be used to manage entities, such as applications, end devices or gateways. For each of these entities, the cli has `create`, `get`, `list`, `update` and `delete` commands.
+
```

Comment:
Also, `contact-info` is part of entity management I think and cli has contact info commands for each entity so probably should mention it here as well.

Reply:
Yes, let's add that when it's actually working 😏